src/Web/Offset/Cache.hs

```diff
@@ -36,50 +41,33 @@ stopReqMutexInt :: MVar (Map WPKey UTCTime) -> WPKey -> IO ()
 stopReqMutexInt activeMV wpKey =
   modifyMVar_ activeMV $ return . Map.delete wpKey
 
-cachingGetRetryInt :: WordpressInt b -> WPKey -> IO (Either StatusCode Text)
+cachingGetRetryInt :: WordpressInt b -> WPKey -> IO Text
 cachingGetRetryInt wp = retryUnless . cachingGetInt wp
 
 cachingGetErrorInt :: WordpressInt b -> WPKey -> IO (Either StatusCode Text)
 cachingGetErrorInt wp wpKey = errorUnless msg (cachingGetInt wp wpKey)
   where msg = "Could not retrieve " <> tshow wpKey
 
-retryUnless :: IO (CacheResult a) -> IO (Either StatusCode a)
+retryUnless :: IO a -> IO a
 retryUnless action =
-  do ma <- action
-     case ma of
-       Successful r -> return $ Right r
-       Abort code -> return $ Left code
-       Retry -> do CC.threadDelay 100000
-                   retryUnless action
-
-errorUnless :: Text -> IO (CacheResult a) -> IO (Either StatusCode a)
+  catch action (\(e :: CacheInProgress) -> CC.threadDelay 100000 >> retryUnless action)
+
```

Comment:
So here I'm caching this `CacheInProgress` exception, but I'm not actually throwing it in `startMutex` yet

(There are no tests that exercise this stuff!)